tests/integration/nets_test_ip/test_podman_compose_nets_test_ip.py

```diff
@@ -0,0 +1,74 @@
+# SPDX-License-Identifier: GPL-2.0
+
+import os
+import unittest
+
+from tests.integration.test_utils import RunSubprocessMixin
+from tests.integration.test_utils import podman_compose_path
+from tests.integration.test_utils import test_path
+
+
+def compose_yaml_path():
+    return os.path.join(os.path.join(test_path(), "nets_test_ip"), "docker-compose.yml")
+
+
+class TestComposeNetsTestIp(unittest.TestCase, RunSubprocessMixin):
+    # test if services retain custom ipv4_address and mac_address matching the subnet provided
+    # in networks top-level element
+    def test_nets_test_ip(self):
+        try:
+            self.run_subprocess_assert_returncode(
+                [
+                    podman_compose_path(),
+                    "-f",
+                    compose_yaml_path(),
+                    "up",
+                    "-d",
+                ],
+            )
+
+            expected_results = [
+                (
+                    "web1",
+                    b"172.19.1.10",
+                    b"02:01:01:00:01:01",
+                    b"172.19.2.10",
+                    b"02:01:01:00:02:01",
+                    b"",
+                ),
+                ("web2", b"", b"", b"172.19.2.11", b"", b"02:01:01:00:02:02"),
```

Comment:
This may match IP address in wrong place. Check for larger substring, e.g. `inet 172.19.2.11 ` (note the space afterwards). Same with mac, prefix should be `link/ether `